paddle/fluid/framework/details/all_reduce_op_handle.h

```diff
@@ -21,20 +21,23 @@
 #include "paddle/fluid/framework/lod_tensor.h"
 #include "paddle/fluid/framework/scope.h"
 #if defined(PADDLE_WITH_CUDA) && !defined(_WIN32)
+#include "paddle/fluid/framework/details/nccl_op_handle.h"
 #include "paddle/fluid/platform/nccl_helper.h"
 #endif
 
 namespace paddle {
 namespace framework {
 namespace details {
 
-class AllReduceOpHandle : public OpHandleBase {
- public:
 #if defined(PADDLE_WITH_CUDA) && !defined(_WIN32)
+class AllReduceOpHandle : public NCCLOpHandleBase {
+ public:
   AllReduceOpHandle(ir::Node *node, const std::vector<Scope *> &local_scopes,
                     const std::vector<platform::Place> &places,
-                    const platform::NCCLContextMap *ctxs);
+                    const platform::MultiNCCLContextMap *ctxs);
 #else
+class AllReduceOpHandle : public OpHandleBase {
```

Comment:
I think that `NCCLOpHandleBase ` is unnecessary.
If I compiled Paddle with GPU, and I want to use it to run a program with DataParallel on CPU. In this case, `AllReduceOpHandle ` inherited from `NCCLOpHandleBase`, it is comfusing.

Reply:
If we don't have ncclhandlebase, the interface it provide move to OphandleBase?